pkg/omitter/reporting_omitter.go

```diff
@@ -0,0 +1,73 @@
+package omitter
+
+import (
+	"sync"
+
+	"github.com/openshift/must-gather-clean/pkg/kube"
+)
+
+type MultiReportingOmitter struct {
+	fileOmitters []FileOmitter
+	k8sOmitters  []KubernetesResourceOmitter
+
+	omittedPathsLock sync.Mutex
+	omittedPaths     []string
+}
+
+func (m *MultiReportingOmitter) OmitPath(path string) (bool, error) {
+	for _, o := range m.fileOmitters {
+		omit, err := o.OmitPath(path)
+		if err != nil {
+			return false, err
+		}
+
+		if omit {
+			m.appendUnderLock(path)
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (m *MultiReportingOmitter) OmitKubeResource(resourceList *kube.ResourceListWithPath) (bool, error) {
+	for _, o := range m.k8sOmitters {
+		omit, err := o.OmitKubeResource(resourceList)
+		if err != nil {
+			return false, err
+		}
+
+		if omit {
+			m.appendUnderLock(resourceList.Path)
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (m *MultiReportingOmitter) Report() []string {
+	m.omittedPathsLock.Lock()
+	defer m.omittedPathsLock.Unlock()
+
+	var copySlice []string
+	for i := 0; i < len(m.omittedPaths); i++ {
+		copySlice = append(copySlice, m.omittedPaths[i])
+	}
+
+	return copySlice
```

Comment:
Why can't we just return `m.omittedPaths` ?
The intention here is to improve the memory efficiency (The new slice memory gets allocated on a heap as the same is getting returned)

Reply:
this is a defensive copy, so a caller is not able to: 
1. change its contents (eg resize on grow)
2. reading can be done outside of the locking

Imagine another thread changing the omittedPathsLock while another thread holds on to the slice internally referenced.